sympy/codegen/fnodes.py

```diff
@@ -0,0 +1,616 @@
+"""
+AST nodes specific to Fortran.
+
+The functions defined in this module allows the user to express functions such as ``dsign``
+as a SymPy function for symbolic manipulation.
+"""
+
+from sympy.core.basic import Basic
+from sympy.core.expr import Expr
+from sympy.core.compatibility import string_types
+from sympy.core.containers import Tuple
+from sympy.core.function import Function
+from sympy.core.numbers import Float, Integer
+from sympy.core.sympify import sympify
+from sympy.codegen.ast import (
+    Attribute, CodeBlock, Declaration, FunctionCall, Node, none, Statement, String,
+    Token, Type, _mk_Tuple, Variable
+)
+from sympy.logic import true, false
+from sympy.utilities.iterables import iterable
+
+
+
+pure = Attribute('pure')
+elemental = Attribute('elemental')  # (all elemental procedures are also pure)
+
+intent_in = Attribute('intent_in')
```

Comment:
Did you intend to write 'indent' instead of 'intent'?

Reply:
Sorry about all the misspellings, in this case though I do mean "intent" (it's a Fortran concept: `intent(in)`, `intent(out)`, `intent(inout)`)